### DISCUSSION TITLE
Esp32 sampler to a drone instrument

### DISCUSSION THREAD

**Discussion by souravlal** [marcel-licence/esp32_midi_sampler#64] [Q&A]
Hi,
I wanted to make a drone instrument which plays 4 notes of a wave file which is uploaded to the PSRAM, with selectable timing interval between the notes and selectable notes to play. But I need all the four notes to have the same length(duration). Will it be possible with setting a high release parameter?
Also is it possible to set a specific value for release and attack(any parameter) in the code itself rather than connecting a MIDI controller(I don't want to connect a MIDI controller).

Where should I look in the code for setting a custom value to attack, release and pitchbend?

Aslo where should I start to add a i2c oled to the mix. I am not a good programmer but I generally understand Arduino codes after a good look at it. But this code seems to be very confusing for me and unable to modify it.
Can anyone please guide me?
Your help will be appreciated.

**Comment by marcel-licence** [comment]
Hi,
it sounds like that you want to play a sequence over and over again like a arpeggiator is that correct?
The release will only change the duration how long the sounds will be audible after the NoteOff. You should use NoteOn and NoteOff with a high decay and sustain to ensure the audio will be audible until the NoteOff will be called.

Do you plan to loop the complete sample?

When loading new samples you should get the correct values:
```
                newPatch->attack = 1.0f;
                newPatch->decay = 1.0f;
                newPatch->sustain = 1.0f;
                newPatch->release = 1.0f;
```

For runtime manipulation you can use: Sampler_SetADSR_Decay, Sampler_SetADSR_Release

What do you want to display on the OLED?
I would recommend to try out to make a little stand-alone project just using the OLED which can be inserted into the sampler later.
You might use the pre-defined I2C pins used for the codec in case they are accessible. It is depending on your hardware.

**Reply by souravlal** [reply]
Thanks for your reply mate.

Please take a look at this video:-👇
https://youtu.be/gLJdOvh-8Hc

This is what I am trying to make. Don't get over confused by the video just take it as a sample. It is a four string instrument(electronic version, there are no strings but only notes) with four of the strings plays at a specific note(For example:- A B C D) in chronological order A to D with a small time delay between them. The string notes gets overlapped each other due to the short time delay between string notes are played as you can hear in the video.

These are the parameters which should be controllable(not every parameter like shown in the video):-

1:- The first string note must be changeable(From A to G)
2:- The main pitch of the instrument should be changeable, ie:- Selecting main pitch changing option(a button for example) should move each string note to one note higher. Fine tuning also necessary.
3:-The tempo ie:- the delay between string notes should be adjustable.

I was wondering whether it is possible to convert this Esp32 midi sampler in to that instrument, and use an oled to display the above 3 parameters.

**Reply by marcel-licence** [reply]
In general I think the ESP32 should be powerful enough to cover functions like that. I am not sure how much work it would be to implement it. Some functions can be implemented by using the sampler as it is and just controlling noteOn/Off and pitch. I will come back later to that topic
